LEGO1/mxdirectx/mxdirectxinfo.h

```diff
@@ -194,19 +194,19 @@ class MxDeviceEnumerate {
 
 	virtual int DoEnumerate(); // vtable+0x00
 
-	BOOL EnumDirectDrawCallback(LPGUID p_guid, LPSTR p_driverDesc, LPSTR p_driverName);
+	BOOL EnumDirectDrawCallback(LPGUID p_guid, LPCSTR p_driverDesc, LPCSTR p_driverName);
 	HRESULT EnumDisplayModesCallback(LPDDSURFACEDESC p_ddsd);
 	HRESULT EnumDevicesCallback(
 		LPGUID p_guid,
-		LPSTR p_deviceDesc,
-		LPSTR p_deviceName,
+		LPCSTR p_deviceDesc,
+		LPCSTR p_deviceName,
 		LPD3DDEVICEDESC p_HWDesc,
 		LPD3DDEVICEDESC p_HELDesc
 	);
 	const char* EnumerateErrorToString(HRESULT p_error);
 	static void BuildErrorString(const char*, ...);
 	static BOOL CALLBACK
-	DirectDrawEnumerateCallback(LPGUID p_guid, LPSTR p_driverDesc, LPSTR p_driverName, LPVOID p_context);
+	DirectDrawEnumerateCallback(LPGUID p_guid, LPCSTR p_driverDesc, LPCSTR p_driverName, LPVOID p_context);
```

Comment:
I think you have to assume callbacks can modify its arguments.

Or you explicitly cast with `(LPDDENUMCALLBACKA)`, but then you lose type safety.

Reply:
Yeah it looks like i sadly can't provide a stricter implementation of the callbacks, I  have reverted that part which should let it compile. But unfortunately that means it doesn't help me on the miniwin implementation.